modules/standard/Path.chpl

```diff
@@ -453,20 +463,24 @@ proc isAbsPath(name: string): bool {
    :rtype: `string`
 */
 proc joinPath(paths: string ...?n): string {
-  var result : string = paths(1); // result variable stores final answer
-  // loop to iterate over all the paths
-  for i in 2..n {
-    var temp : string = paths(i);
-    if temp.startsWith('/') {
-      result = temp;
-    }
-    else if result.endsWith('/') {
-      result = result + temp;
-    }
-    else {
-      result = result + "/" + temp;
-    }
-  }
+  var result: string;
+
+  for path in paths do
+    joinPathComponent(path, result);
+
+  return result;
+}
+
+// This overload is private for now, needed for relPath.
```

Comment:
nit: Still good to use doc-string `/* ... */` comments even for non-chpldoc comments.